benchmarks/mathvista/mathvista.py

```diff
@@ -0,0 +1,160 @@
+    choices = record["choices"]
+    labels = [chr(i) for i in range(ord("a"), ord("a") + len(choices))]
+
+    choices = dict(zip(labels, choices))
+
+    answer = record["answer"]
+    target = list(choices.values()).index(answer)
+    target = chr(ord("A") + int(target))
+
+    return target
+
+
+def record_to_sample(record: dict) -> Sample:
+    message = [
+        ChatMessageUser(
+            content=[
+                ContentText(text=record["question"])
+                if record["question_type"] == "multi_choice"
+                else ContentText(text=record["query"]),
+                ContentImage(image=record["image"]),
+            ]
+        )
+    ]
+    if record["question_type"] == "multi_choice":
+        return Sample(
+            input=message,
+            choices=record["choices"],
+            target=get_multi_choice_as_letter(record),
+            id=record["pid"],
+            metadata={
+                "question_type": record["question_type"],
+                "answer_type": record["answer_type"],
+                **record["metadata"],
+            },
+            files={f"image:{record['image']}": record["image"]},
+        )
+
+    elif record["question_type"] == "free_form":
+        return Sample(
+            input=message,
+            target=record["answer"],
+            id=record["pid"],
+            metadata={
+                "precision": record["precision"],
+                "question_type": record["question_type"],
+                "answer_type": record["answer_type"],
+                **record["metadata"],
+            },
+            files={f"image:{record['image']}": record["image"]},
+        )
+
+
+@solver
+def multi_choice_or_free_form_solver() -> Solver:
+    async def solve(state: TaskState, generate: Generate) -> TaskState:
+        if state.metadata["question_type"] == "multi_choice":
+            state.user_prompt.text = prompt(
+                question=state.user_prompt.text,
+                choices=state.choices,
+                template=SINGLE_ANSWER_TEMPLATE,
+            )
+
+            return await generate(state)
+
+        elif state.metadata["question_type"] == "free_form":
+            state.user_prompt.text = FREEFORM_TEMPLATE.format(
+                question=state.user_prompt.text
+            )
+            return await generate(state)
+
+    return solve
+
+
+@scorer(metrics=[accuracy(), stderr()])
+def multi_choice_or_free_form_scorer():
+    async def score(state: TaskState, target: Target) -> Score:
+        match = re.search(
+            ANSWER_PATTERN_LETTER
+            if state.metadata["question_type"] == "multi_choice"
+            else ANSWER_PATTERN_LINE,
+            state.output.completion,
+            re.IGNORECASE,
+        )
+        if match:
+            groups = match.groups()
+            found_match = match_first(matches=groups, target=target, ignore_case=True)
+
+            if found_match is None and len(groups) == 1:
```

Comment:
Why do we need `len(groups) == 1` here? Is this to mitigate if the model has multiple 'ANSWER: $ANSWER's in its response?

Reply:
Yup, that's right!